### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by FarhanMobashir**
following are the things added in this pr

- adds some toast for various action according to error and success
- adds basic form validation
- add timing to the api
- adds skeleton loader
- adds the category navigation and category filter
- added 404 page

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -30,6 +30,7 @@
   </head>
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
+    <div id="modal"></div>
```
Comment (Hamdrive) [inline public/index.html:33]: Shouldn't we ideally not alter the html file? it should only have root div within body

File: src/components/Input.js [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+    />
+    {/* validation text */}
+    {/* <small
+      class={`validation-message ${validationStatus ? "green-5" : "red-5"}`}
+    >
+      <i
+        class={`uil ${validationStatus ? "uil-exclamation-circle" : "uil-check"}`}
+      ></i>
+      {validationText}
+    </small> */}
```
Comment (Hamdrive) [inline src/components/Input.js:25]: Good idea to not push these changes to PR but keep them on local

File: src/components/Toast.js [hunk 3]
```diff
@@ -1,39 +1,42 @@
...
+        <i
+          className={`toast-icon uil 
+        ${
+          type === "success"
+            ? "uil-check"
+            : type === "danger"
+            ? "uil-exclamation-circle"
+            : type === "info"
+            ? "uil-shield-exclamation"
+            : ""
```
Comment (Hamdrive) [inline src/components/Toast.js:26]: Interesting ternary implementation for different toast styles

File: src/global.css [hunk 4]
```diff
@@ -398,15 +398,14 @@
 .product-listing-container {
   padding: 1% 2%;
   display: flex;
-  align-items: center;
+  /* align-items: center; */
```
Comment (Hamdrive) [inline src/global.css:401]: Cleanup unused commented code

File: src/helpers/buildApiHooks.js [hunk 5]
```diff
@@ -116,7 +116,7 @@ export const buildHooks = (queryArray, baseQuery, dispatchFn) => {
           body: JSON.stringify(body),
         })
           .then((res) => {
-            console.log(res);
+            // console.log(res);
```
Comment (Hamdrive) [inline src/helpers/buildApiHooks.js:119]: Cleanup console.logs

**Review by Hamdrive — COMMENTED** [review]
Good work creating individual components, especially for wildcard route (Empty state)
Liked the use of toast for handling products
Learnt a new way to implement modal using useRef hook, will apply this and other learning in my other projects

**Review by GuptaEesh — COMMENTED** [review]
Can use barrel imports to ease importing files.
Overall the code looks good.
The comments made by the other reviewer should be implemented.
Other than that looks good to merge

File: public/index.html (hunk 1 above)
Comment (FarhanMobashir) [reply public/index.html:33]: Actually I am using portal for the modal and that's why I am doing like this

**Pull request merged**